### PR TITLE
chore: cleanup organization of the process managers (on top of #416)

### PR DESCRIPTION
**Pull request by doudou**
Depends on:
- [ ] https://github.com/rock-core/tools-syskit-log/pull/49
- [ ] https://github.com/rock-gazebo/simulation-rock_gazebo/pull/55
- [ ] https://github.com/rock-gui/gui-orogen-qt_base/pull/2

On top of #416

The next PR will build on top of this to import Orocos::Process into syskit and clean that up.

### CODE REVIEW

File: lib/syskit/process_managers/in_process/manager.rb [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+                    name_mappings.each { process.map_name(_1, _2) }
+                end
+
+                # Requests that the process server moves the log directory at +log_dir+
+                # to +results_dir+
+                def save_log_dir(log_dir, results_dir); end
+
+                # Creates a new log dir, and save the given time tag in it (used later
+                # on by save_log_dir)
+                def create_log_dir(log_dir, time_tag, metadata = {}); end
```
Comment (wvmcastro) [inline lib/syskit/process_managers/in_process/manager.rb:92]: should this raise `NotImplemented` ?

File: lib/syskit/process_managers/in_process/manager.rb [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+
+                # Waits for processes to terminate. +timeout+ is the number of
+                # milliseconds we should wait. If set to nil, the call will block until
+                # a process terminates
+                #
+                # Returns a hash that maps deployment names to the Status
+                # object that represents their exit status.
+                def wait_termination(_timeout = nil)
```
Comment (wvmcastro) [inline lib/syskit/process_managers/in_process/manager.rb:100]: The comment suggests that `_timeout` is useful but the implementation ignores it. Is it here just for some api consistence ?

File: lib/syskit/process_managers/in_process/manager.rb [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+                        true
+                    end
+                    dead_processes
+                end
+
+                def wait_running(*process_names)
+                    result = {}
+                    process_names.each do |name|
+                        if (p = processes[name])
+                            result[name] = { iors: p.wait_running(0) }
```
Comment (wvmcastro) [inline lib/syskit/process_managers/in_process/manager.rb:115]: iors ?

**Review by jhonasiv — APPROVED** [review]

File: lib/syskit/process_managers/in_process/manager.rb (hunk 2 above)
Comment (doudou) [reply lib/syskit/process_managers/in_process/manager.rb:100]: Yes. Consistence with the {Orocos::Process} API. The end goal is to remove all that old unused stuff by making all of this (1) syskit-specific and (2) private.

In Syskit, `timeout` is always zero.

File: lib/syskit/process_managers/in_process/manager.rb (hunk 3 above)
Comment (doudou) [reply lib/syskit/process_managers/in_process/manager.rb:115]: IOR is the string representation of an address in CORBA. It is what allows us to create Orocos::TaskContext to control orogen components.

File: lib/syskit/process_managers/in_process/manager.rb (hunk 1 above)
Comment (doudou) [reply lib/syskit/process_managers/in_process/manager.rb:92]: No, it's a no-op. Will be removed later on once the process managers do not depend anymore on anything from `tools/orocos.rb`

**Review by wvmcastro — APPROVED** [review]

**Pull request merged**